chronos/transports/stream.nim

```diff
@@ -1135,16 +1145,19 @@ when defined(windows):
     if server.local.family in {AddressFamily.IPv4, AddressFamily.IPv6}:
       # TCP Sockets part
       var loop = getThreadDispatcher()
-      server.asock = try: createAsyncSocket(server.domain, SockType.SOCK_STREAM,
-                                       Protocol.IPPROTO_TCP)
-      except CatchableError as exc:
-        retFuture.fail(exc)
-        return retFuture
+      server.asock =
+        try:
+          createAsyncSocket(server.domain, SockType.SOCK_STREAM,
+                            Protocol.IPPROTO_TCP)
+        except CatchableError as exc:
+          retFuture.fail(exc)
+          return retFuture
 
       if server.asock == asyncInvalidSocket:
         let err = osLastError()
-        if int32(err) == ERROR_TOO_MANY_OPEN_FILES:
-          retFuture.fail(getTransportTooManyError())
+        case int(err)
+        of ERROR_TOO_MANY_OPEN_FILES, WSAENOBUFS, WSAEMFILE:
```

Comment:
I wonder who ever had the idea that we need 3 things to specify "out of files" haha. But good catch.

Actually makes me think whether EAGAIN / EWOULDBLOCK should also both be handled (on some weird platforms they are not the same value, and currently we only handle EAGAIN).